services/user.py

```diff
@@ -0,0 +1,49 @@
+from django.contrib.auth import get_user_model
+
+
+def create_user(username: str,
+                password: str,
+                email: str = None,
+                first_name: str = None,
+                last_name: str = None) -> None:
+    user_model = get_user_model()
+    user = user_model(username=username)
+    user.set_password(password)
+
+    if email:
+        user.email = email
+    if first_name:
+        user.first_name = first_name
+    if last_name:
+        user.last_name = last_name
+
+    user.save()
+    return user

```

Comment:
In the `create_user` function, you're calling `user.save()` unconditionally. This means that even if no changes were made to the user, the `save()` method will still be called. This is unnecessary and can degrade the performance of your application. Instead, you should only call `save()` if at least one field was updated. You can achieve this by adding a boolean variable that tracks whether any field was updated. If this variable is `True` at the end of the function, call `save()`. Otherwise, don't call it.